### DISCUSSION TITLE
Run step based on which previous step failed

### DISCUSSION THREAD

**Discussion by JeffersonBledsoe** [community/community#26969] [Actions]
<p>I’m planning on writing a workflow which executes the unit, integration and E2E tests as separate steps. As part of this, I would like to change what artifacts are uploaded for a failed test run based on which testing step failed. Is it possible for a step to know which step failed? E.g. unit tests should upload just a summary of the unit tests, whilst the E2E tests may include screenshots of the failed tests?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/jeffersonbledsoe">@jeffersonbledsoe</a> , </p>
<p>You could use <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#steps-context" rel="nofollow noopener">steps.&lt;step id&gt;.outcome</a> to get the step result.  And you could use if conditional in the upload artifacts step, add always() in addition to let the step to run even if previous step failed.  </p>
<p>if: always() &amp;&amp; steps.step2.outcome == ‘failure’</p>
<pre><code>jobs:
  build:
    runs-on: ubuntu-latest
    steps:
     - name: step1
       run: echo "hello 1"
       id: step1
       continue-on-error: true
     - name: step2
       run: |
          echo "hello 2"
          fff
       id: step2
       #continue-on-error: true
     - name: step3
       if: always() &amp;&amp; steps.step2.outcome == 'failure'
       run: echo "hello 3"
       id: step3
</code></pre>

**Comment by JeffersonBledsoe** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> Works great, thanks for the help!</p>
